src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -485,7 +510,7 @@ private Settings getSettings() {
      *
      * @param timeout  The timeout for the listener in milliseconds. A timeout of 0 means no timeout.
      */
-    public void startActionListener(int timeout) {
+    public static void startActionListener(int timeout) {
```

Comment:
Why do we have to change it to `static`?

Reply:
@owaiskazi19 The normal answer to that question is that it doesn't affect any instance variables so it would help the compiler (potentially) optimize.  However...

_Why_ doesn't it affect any instance variables?  It looks like we instantiate an `ActionListener` on the next line, tell it to run, and then promptly forget about it, so we can't get any info from it again, and it is eligible for garbage collection unless one of its internal variables has a reference retained.

Which it doesn't.  It does open a `ServerSocket` which looks internally like it starts a thread... which dies on handshake or after an hour, but that's not attached to any object so it looks like it just floats out there for a while, keeping the socket alive.

There doesn't appear to be any way to prevent multiple action listeners from simultaneously executing, which in the case of a static method is possibly a good thing, as each of multiple extensions may start one up... except that after starting them up they have no reference to it anymore.

All this is to say this should probably _not_ be static, but there's a bigger problem of not keeping track of the action listener at all. Shouldn't it be an instance variable and shouldn't this method only start it if it hasn't already been started?  Or as part of a constructor?

I realize this problem existed before this PR, but I suspect the failure will never be triggered (or if it is, nobody will get the result of it).

Reply:
I don't remember changing it to static, so I'm guessing it's just something that happened with merge conflicts.

Reply:
That makes sense @dbwiddis! Thanks for the detailed explanation.